src/gui/widgets/FloatModelEditorBase.cpp

```diff
@@ -278,8 +282,79 @@ void FloatModelEditorBase::paintEvent(QPaintEvent *)
 	p.setPen(foreground);
 	p.setBrush(foreground);
 	p.drawRect(QRect(r.topLeft(), QPoint(r.width() * percentage, r.height())));
+	drawAutoHighlight(&p);
 }
 
+std::vector<InteractiveModelView::ModelShortcut> FloatModelEditorBase::getShortcuts()
+{
+	std::vector<InteractiveModelView::ModelShortcut> shortcuts = {
+		InteractiveModelView::ModelShortcut(Qt::Key_C, Qt::ControlModifier, 0, QString(tr("Copy value")), false),
+		InteractiveModelView::ModelShortcut(Qt::Key_C, Qt::ControlModifier, 1, QString(tr("Link widget")), false),
+		InteractiveModelView::ModelShortcut(Qt::Key_V, Qt::ControlModifier, 0, QString(tr("Paste value")), false)
+	};
+	return shortcuts;
+}
+
+void FloatModelEditorBase::processShortcutPressed(size_t shortcutLocation, QKeyEvent* event)
```

Comment:
It would be nice if the `size_t shortcutLocation` parameter could be `const ModelShortcut&` instead.

Reply:
I would like to keep the `size_t` because I think it is easier to use in this case (with `switch`) and it could be more optimized (checking if `size_t` == `size_t` instead of `ModelShortcut` == `shortcutArray[index]`).